### PR TITLE
Fixes issues with the debian and broken SuSe init scripts

### PR DESCRIPTION
**Pull request by svanoort**
Specifically [JENKINS-31348](https://issues.jenkins-ci.org/browse/JENKINS-31348) (broken SuSe init script due to abuse of the /proc system to find the Jenkins process) and the Debian script silently allowing Jenkins failures to pass [JENKINS-31064](https://issues.jenkins-ci.org/browse/JENKINS-31064). 

Builds on top of https://github.com/jenkinsci/packaging/pull/26

@reviewbybees

### CODE REVIEW

File: deb/build/debian/jenkins.init [hunk 1]
```diff
@@ -207,10 +221,13 @@ case "$1" in
     case "$?" in
       0|1)
         do_start
+        START_STATUS="$?"
+        sleep 1
+        do_check_started_ok "$START_STATUS"
         case "$?" in
```
Comment (oleg-nenashev) [inline deb/build/debian/jenkins.init]: :bug: (?) case should take <code>S$TART_STATUS</code>

Comment (oleg-nenashev) [inline deb/build/debian/jenkins.init]: Nope, please ignore

**Comment by rsandell** [comment]
:bee:

**Comment by oleg-nenashev** [comment]
:bee:

**Comment by svanoort** [comment]
@reviewbybees done.

**Comment by amuniz** [comment]
:bee: (with my limited shell script knowledge :smile:)

File: suse/build/SOURCES/jenkins.init.in [hunk 2]
```diff
@@ -106,9 +106,9 @@ case "$1" in
                 else
                        HOME=$JENKINS_HOME startproc -n $JENKINS_NICE -s -e -l /var/log/@@ARTIFACTNAME@@.rc -u "$JENKINS_USER" -p "$JENKINS_PID_FILE" $JAVA_CMD $PARAMS
                 fi
-                JPROC=$( find /proc -maxdepth 2 -user $JENKINS_USER -name exe -lname "*/bin/java" )
+                JPROC=$( pgrep java -U jenkins )
```
Comment (jtnord) [inline suse/build/SOURCES/jenkins.init.in]: does the package not need a dependency on whatever contains `pgrep`?

Comment (jtnord) [inline suse/build/SOURCES/jenkins.init.in]: ` $JENKINS_USER` not hard coded `jenkins`

**Comment by jtnord** [comment]
:bug: on the hard coded user name

File: rpm/build/SPECS/jenkins.spec [hunk 3]
```diff
@@ -32,7 +32,7 @@ BuildRoot:	%{_tmppath}/build-%{name}-%{version}
 #    that there's no such packages. JRE/JDK RPMs from java.sun.com
 #    do not have this virtual package declarations. So for now,
 #    I'm dropping this requirement.
-# Requires:	java >= 1:1.6.0
+# Requires:	java >= 1:1.6.0, procps
```
Comment (jtnord) [inline rpm/build/SPECS/jenkins.spec]: 1.6 and commented out???

File: suse/build/SPECS/jenkins.spec [hunk 4]
```diff
@@ -29,7 +29,7 @@ BuildRoot:	%{_tmppath}/build-%{name}-%{version}
 # provides >= 1.6.0 must specify the epoch, "java >= 1:1.6.0".
 #
 # java-1_6_0-sun provides this at least
-Requires:	java >= 1.6
+Requires:	java >= 1.6, procps
```
Comment (jtnord) [inline suse/build/SPECS/jenkins.spec]: 1.6

**Comment by jtnord** [comment]
:bee: as discussed the java dependencies are wrong - but that can be tackled in a later PR - as it is unclear how you can correctly differentiate OpenJDK vs OracleJDK at the correct versions.

**Comment by svanoort** [comment]
@jtnord Interesting side point, the SUSE installer will actually resolve to pulling in openjdk 1.7 by default with that dependency.  I can't actually find a 1.6 JRE for them in packagesets using zypper searches (but I'm not that SUSE fluent).

**Comment by svanoort** [comment]
merging as pseudo-maintainer for integrated testing

**Pull request merged**